gems/uc3-dmp-external-api/lib/uc3-dmp-external-api/client.rb

```diff
@@ -25,11 +25,12 @@ def call(url:, method: :get, body: '', basic_auth: {}, additional_headers: {}, t
         opts = _options(body:, basic_auth:, additional_headers:, timeout:, logger:)
         resp = HTTParty.send(method.to_sym, uri, opts)
 
-        unless [200, 201].include?(resp.code)
-          msg = "status: #{resp&.code}, body: #{resp&.body}"
-          raise ExternalApiError, "#{format(MSG_ERROR_FROM_EXTERNAL_API, url:)} - #{msg}"
+        if [200, 201].include?(resp.code.to_i)
+          resp.body.nil? || resp.body.empty? ? nil : _process_response(resp:)
+        else
+          puts "The response code is not 200 or 201"
```

Comment:
Please change this to use the logger: `logger&.warn(message: "#{format(MSG_ERROR_FROM_EXTERNAL_API, url:)} - HTTP status code #{rest.code}")`

Reply:
and then you should return something here. Since we return a null value for other scenarios above, you should add a final line here that reads `nil`

Reply:
Thanks Brian. I made the update.